src/topology/fiber_bundle/basic.lean

```diff
@@ -134,130 +142,164 @@ Fiber bundle, topological bundle, structure group
 variables {ι : Type*} {B : Type*} {F : Type*}
 
 open topological_space filter set bundle
-open_locale topological_space classical
+open_locale topological_space classical bundle
+
+/-! ### General definition of fiber bundles -/
+
+section fiber_bundle
 
-/-! ### General definition of topological fiber bundles -/
+variables (F) [topological_space B] [topological_space F] (E : B → Type*)
 
-section topological_fiber_bundle
+/-! ### Fiber bundles -/
 
-variables (F) {Z : Type*} [topological_space B] [topological_space F] {proj : Z → B}
-variable [topological_space Z]
+variables [topological_space (total_space E)] [∀ b, topological_space (E b)]
 
-/-- A topological fiber bundle with fiber `F` over a base `B` is a space projecting on `B`
+/-- A (topological) fiber bundle with fiber `F` over a base `B` is a space projecting on `B`
 for which the fibers are all homeomorphic to `F`, such that the local situation around each point
 is a direct product. -/
-def is_topological_fiber_bundle (proj : Z → B) : Prop :=
-∀ x : B, ∃e : trivialization F proj, x ∈ e.base_set
+class fiber_bundle :=
+(total_space_mk_inducing [] : ∀ (b : B), inducing (@total_space_mk B E b))
+(trivialization_atlas [] : set (trivialization F (π E)))
```

Comment:
Now that trivializations are only used in a Sigma type situation, instead of parameterizing them by the projection I wonder if we shouldn't parameterize them just by the type, i.e., write `trivialization F E` instead of `trivialization F (π E)`. Probably in another PR if you think it's a good idea, though.

Reply:
For myself I'm not sure.  It's certainly more elegant to have the trivialization concept be adapted to the definition chosen for fibre bundles.  But the more general trivializations are still (as of this PR) used in a couple of places:
- for `is_trivial_fiber_bundle` as we are discussing in the other comment
- to say that a covering map admits local trivializations

@fpvandoorn, what do you think?  I think you were in favour of keeping the more general version before.

Reply:
I still think it's nice to have some tools that allow you to say things similar to `𝕊³` is a fiber bundle over `𝕊²`. Even if we don't allow this exact statement for fiber bundles, it might be useful to keep allowing it for trivializations...